pkg/watcher/certificate.go

```diff
@@ -0,0 +1,263 @@
+	if k8serrors.IsNotFound(err) {
+		c.logger.Info("Certificate does not exist", "Certificate", c.certificateName)
+		return false, nil
+	} else if err != nil {
+		c.logger.Info("Could not fetch certificate from local Cluster")
+		return false, err
+	}
+	return true, nil
+}
+
+func (c *Certificate) createCertificate(
+	ctx context.Context, certNamespace string,
+	subjectAltName *SubjectAltName,
+) error {
+	// Duration Default 90 days
+	// RenewBefore default 2/3 of Duration
+	issuer, err := c.getIssuer(ctx)
+	if err != nil {
+		c.logger.Error(err, "Error getting Issuer")
+		return err
+	}
+	c.logger.Info("Issuer found", "issuer", issuer)
+
+	cert := v1.Certificate{
+		ObjectMeta: apimachinerymetav1.ObjectMeta{
+			Name:      c.certificateName,
+			Namespace: certNamespace,
+		},
+		Spec: v1.CertificateSpec{
+			DNSNames:       subjectAltName.DNSNames,
+			IPAddresses:    subjectAltName.IPAddresses,
+			URIs:           subjectAltName.URIs,
+			EmailAddresses: subjectAltName.EmailAddresses,
+			SecretName:     c.secretName, // Name of the secret which stored certificate
+			SecretTemplate: &v1.CertificateSecretTemplate{
+				Labels: LabelSet,
+			},
+			IssuerRef: metav1.ObjectReference{
+				Name: issuer.Name,
+			},
+			IsCA: false,
+			Usages: []v1.KeyUsage{
+				v1.UsageDigitalSignature,
+				v1.UsageKeyEncipherment,
+			},
+			PrivateKey: &v1.CertificatePrivateKey{
+				RotationPolicy: privateKeyRotationPolicy,
+				Encoding:       v1.PKCS1,
+				Algorithm:      v1.RSAKeyAlgorithm,
+			},
+		},
+	}
+
+	return c.kcpClient.Create(ctx, &cert)
+}
+
+func (c *Certificate) getSubjectAltNames(config *SkrChartManagerConfig) (*SubjectAltName, error) {
+	if domain, ok := c.kyma.Annotations[DomainAnnotation]; ok {
+		if domain == "" {
+			return nil, fmt.Errorf("Domain-Annotation of KymaCR %s is empty", c.kyma.Name) //nolint:goerr113
+		}
+
+		resourceName := ResolveSKRChartResourceName(WebhookCfgAndDeploymentNameTpl,
+			client.ObjectKeyFromObject(c.kyma))
+		namespace := c.kyma.Namespace
+		svcSuffix := []string{"svc.cluster.local", "svc"}
+		dnsNames := []string{domain}
+
+		for _, suffix := range svcSuffix {
+			dnsNames = append(dnsNames, fmt.Sprintf("%s.%s.%s", resourceName, namespace, suffix))
+		}
+
+		if config.WatcherLocalTestingEnabled {
+			dnsNames = append(dnsNames, []string{"localhost", "127.0.0.1", "host.k3d.internal"}...)
+		}
+
+		return &SubjectAltName{
+			DNSNames: dnsNames,
+		}, nil
+	}
+	return nil, fmt.Errorf("kymaCR %s does not contain annotation '%s' with specified domain", //nolint:goerr113
+		c.kyma.Name, DomainAnnotation)
+}
+
+func (c *Certificate) getIssuer(ctx context.Context) (*v1.Issuer, error) {
+	issuerList := &v1.IssuerList{}
+	err := c.kcpClient.List(ctx, issuerList, &client.ListOptions{
+		LabelSelector: k8slabels.SelectorFromSet(LabelSet),
+		Namespace:     c.kyma.Namespace,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not list cert-manager issuer %w", err)
+	}
+	if len(issuerList.Items) == 0 {
```

Comment:
If there is more than one issuer, you are missing a warning log or an error here.